### ISSUE TITLE
Flaky Test Tracker

### ISSUE THREAD

**Issue by russjones** [gravitational/teleport#9492]
#### Process

1. Start with the assumption the test is correct and is highlighting a bug in Teleport.
2. Run multiple parallel unit or integration tests to reproduce.
3. Attempt to fix the test.
4. Propose quarantine.

#### Unit Tests

* ~_Frequently fails._ `github.com/gravitational/teleport/lib/service.TestTeleportProcess_reconnectToAuth`~
* ~_Frequently fails._ `github.com/gravitational/teleport/lib/srv/regular.TestClientDisconnect`~
* ~_Frequently fails._ `github.com/gravitational/teleport/lib/cache.TestCache_Backoff`~
* ~`github.com/gravitational/teleport/lib/srv/regular.TestProxyReverseTunnel`~
* ~`github.com/gravitational/teleport/lib/auth.TestAPILockedOut`~
* `github.com/gravitational/teleport/lib/auth.TestAPI`
* _Often fails locally_ `github.com/gravitational/teleport/lib/auth.TestTiming` (also reported in #4653)

#### Integration

* ~_Frequently fails._ [`TestIntegrations/TwoClustersTunnel`]~(https://github.com/gravitational/teleport/issues/9655)
* [`TestIntegrations/Disconnection`](https://github.com/gravitational/teleport/issues/13699)
* [`TestHSMDualAuthRotation`](https://github.com/gravitational/teleport/issues/13635)
* [`TestHSMMigrate`](https://github.com/gravitational/teleport/issues/14172)
* `TestIntegrations/MultiplexingTrustedClusters`
* ~[`TestIntegrations/RotateTrustedClusters`](https://github.com/gravitational/teleport/issues/13569)~

### Metrics

_Trailing 7-day pass rate for unit and integration tests._

* Week of January 10th. Unit 67%, Integration 56%

### Proposed for Quarantine

_This section is for tests that provide business value but are inherently flaky due to a dependence on time and an external resource (like CPU or network). For example, a test that waits for an event to occur and times out if the event does not occur after some time._

_Quarantined tests will be triaged by @russjones weekly and potentially serialized and put into a retry loop._

* https://github.com/gravitational/teleport/issues/9491
* `lib/auth.PasswordSuite.TestTiming` requires exists/not exists tests to be within 10% of eachother

### Fixed

* https://github.com/gravitational/teleport/pull/9316
* https://github.com/gravitational/teleport/pull/9326
* https://github.com/gravitational/teleport/pull/9119
* https://github.com/gravitational/teleport/pull/9118
* https://github.com/gravitational/teleport/pull/9117
* https://github.com/gravitational/teleport/pull/8888
* https://github.com/gravitational/teleport/pull/8643
* https://github.com/gravitational/teleport/pull/8608
* https://github.com/gravitational/teleport/pull/9316
* https://github.com/gravitational/teleport/pull/8744

**Comment by fspmarshall** [comment]
Found a race in `TwoClustersTunnel` which I believe is the cause of this error:

```
failed connecting to node localhost. remote cluster &#34;site-A&#34; is not found
```

I'm not sure if this is the only problem with `TwoClustersTunnel`, but I'll push a fix and we can watch to see if its flakiness goes down.

**Comment by tcsc** [comment]
I see this quite often, too: 
```
integration_test.go:1510: 
Error Trace:	integration_test.go:1510
                        integration_test.go:1388
Error:      	Received unexpected error:
       	            	connection error: desc = "transport: Error while dialing failed to dial: failed connecting to node . invalid format for proxy request: unknown cluster \"site-A\"\n"
```

**Comment by zmb3** [comment]
Is this a duplicate of #4653? Should we combine them?

**Comment by russjones** [comment]
@zmb3 Yeah, I think we should close that one and merge things into this one.

**Comment by zmb3** [comment]
`TwoClustersTunnel` often fails for me with:

```
failed connecting to node localhost. database is closed
```

Occasionally I get a similar error that cays `cache is closed` instead of database.

Interestingly:
- I can only reproduce when I run with the race detector enabled.
- If I remove the section of the test that stops site A and restarts it, I see completely different errors

At some point in the tests, I start seeing tons of `Uploader scan failed` errors. I have a feeling the vast majority of this code is failing to clean up properly and is removing directories while they are still in use.

**Comment by rosstimothy** [comment]
Opened #9516 to address:
- `github.com/gravitational/teleport/lib/service.TestTeleportProcess_reconnectToAuth`
- `github.com/gravitational/teleport/lib/service.TestResourceWatcher_Backoff`
- `github.com/gravitational/teleport/lib/cache.TestCache_Backoff`

**Comment by russjones** [comment]
For `TestIntegrations/TwoClustersTunnel`: https://github.com/gravitational/teleport/issues/9655

**Comment by russjones** [comment]
A good way to reproduce issue is using `while`. For example:

```
$ while go test . -run TestAccessMongoDB -count=1 -race; do :; done
```

**Comment by tcsc** [comment]
I have a similar script I use called `untilfail`: 

```bash
#!/bin/bash

COUNT=1
while "$@"; do COUNT=$((COUNT + 1)); done

echo Ran $COUNT times
```
Then I can pass it an arbitrary command line:
```
$ untilfail go test ./integration -run TestSomethingOrOther -race
```

**Comment by zmb3** [comment]
I've started seeing `TestSSHConfigConnectWithOpenSSHClient` failures. Not sure if it's just my environment or not, but figured I'd put a note here. Spent 30 minutes or so debugging and I don't know what's going on.

```
zmb@localhost: Permission denied (publickey).
ERROR: exit status 255
```

**Comment by zmb3** [comment]
Here's a new one that I ran into on GCB. Can't seem to repro locally.

```
OUTPUT github.com/gravitational/teleport/lib/utils.TestFnCacheSanity
===================================================
=== RUN   TestFnCacheSanity
    fncache_test.go:145: 
        	Error Trace:	fncache_test.go:145
        	            				fncache_test.go:71
        	Error:      	Max difference between 10.461059975 and 9 allowed is 1, but difference was 1.4610599749999995
        	Test:       	TestFnCacheSanity
        	Messages:   	ttl=40ms, delay=0s, desc="non-blocking"
--- FAIL: TestFnCacheSanity (1.53s)
===================================================
```

**Comment by zmb3** [comment]
@r0mant have you seen this one before?

```
2022-03-14T20:24:44Z ERRO [AUTH:2]    PID: 55274 Failed to bind to address 127.0.0.1:32963: listen tcp 127.0.0.1:32963: bind: address already in use, exiting. service/service.go:1330
    helpers_test.go:145: 
        	Error Trace:	helpers_test.go:145
        	            				resource_command_test.go:138
        	Error:      	Received unexpected error:
        	            	listen tcp 127.0.0.1:32963: bind: address already in use
        	Test:       	TestDatabaseResource
--- FAIL: TestDatabaseResource (3.35s)
```

**Comment by zmb3** [comment]
Another one: `TestDatabaseRootLeafIdleTimeout/leaf_role_with_idle_timeoutz`

```
    db_integration_test.go:371: 
        	Error Trace:	db_integration_test.go:371
        	Error:      	An error is expected but got nil.
        	Test:       	TestDatabaseRootLeafIdleTimeout/leaf_role_with_idle_timeout
```

**Comment by zmb3** [comment]
`TestLockWatcherStale` is back on v8. @rosstimothy didn't we fix this one too?

```
=== CONT  TestLockWatcherStale
    watcher_test.go:502: 
        	Error Trace:	watcher_test.go:502
        	            				watcher_test.go:444
        	Error:      	An error is expected but got nil.
        	Test:       	TestLockWatcherStale
--- FAIL: TestLockWatcherStale (2.27s)
2022-03-18T17:08:30Z DEBU             Closed, returning from watch loop. resource-kind:lock services/watcher.go:214
```

**Comment by zmb3** [comment]
```
    testing.go:967: TempDir RemoveAll cleanup: unlinkat /tmp/TestHandleConnection570641151/001/log/upload/streaming/default: directory not empty
--- FAIL: TestHandleConnection (5.38s)
```

**Comment by zmb3** [comment]
`TestTerminateNotifier` is flaky due to the following:

```
if b == 0x03 {
	g.mu.Lock()
	if !g.on && !g.closed {
		select {
		case g.terminateNotifier <- struct{}{}:
		default:
		}
	}
	g.mu.Unlock()
	break
}
```

We only write to `g.terminateNotifier` is there is a pending channel read in another goroutine. Since the test executes a background write and then starts waiting on the channel, the ctrl-c can be detected before the channel read.

@xacrimon I'm not familiar enough with the semantics here to make a change, and this code has proven fragile over the past couple of weeks. Maybe should consider a buffered channel or removing the default case, ensuring we don't miss a close notification.

**Comment by xacrimon** [comment]
PR up with reasoning in https://github.com/gravitational/teleport/pull/11687

**Comment by zmb3** [comment]
Looks like `TestTSHSSH` has some racy cleanup behavior:

```
=== CONT  TestTSHSSH
    testing.go:967: TempDir RemoveAll cleanup: unlinkat /tmp/TestTSHSSH372711066/002/log/upload/streaming/default: directory not empty
--- FAIL: TestTSHSSH (15.59s)
===================================================
make: *** [test-go] Error 1
```

cc @smallinsky (looks like you edited this section recently)

Also note, this test appears to delete `~/.tsh`. We should use `t.TempDir()` and `$TELEPORT_HOME` to avoid clobbering local state for devs.

**Comment by zmb3** [comment]
```
FAIL: services_test.go:163: ServicesSuite.TestSemaphoreFlakiness

services_test.go:164:
    s.suite.SemaphoreFlakiness(c)
/workspace/lib/services/suite/suite.go:1185:
    c.Fatalf("Lost semaphore lock: %v", lock.Wait())
... Error: Lost semaphore lock: uh-oh!
```

This one's a real pain because the services tests run under a giant "check" test suite, making it hard to just run the one flaky test in a loop to reproduce.

**Comment by greedy52** [comment]
> `TestLockWatcherStale` is back on v8. @rosstimothy didn't we fix this one too?
> 
> ```
> === CONT  TestLockWatcherStale
>     watcher_test.go:502: 
>         	Error Trace:	watcher_test.go:502
>         	            				watcher_test.go:444
>         	Error:      	An error is expected but got nil.
>         	Test:       	TestLockWatcherStale
> --- FAIL: TestLockWatcherStale (2.27s)
> 2022-03-18T17:08:30Z DEBU             Closed, returning from watch loop. resource-kind:lock services/watcher.go:214
> ```

I just run into this on master. Seems a race condition. I will make a fix.

**Comment by hatched** [comment]
```
--- FAIL: TestWebSessionsRenewDoesNotBreakExistingTerminalSession (0.78s)
    apiserver_test.go:4174:
        	Error Trace:	apiserver_test.go:4174
        	            				apiserver_test.go:3125
        	Error:      	Received unexpected error:
        	            	EOF
        	Test:       	TestWebSessionsRenewDoesNotBreakExistingTerminalSession
    apiserver_test.go:4109:
        	Error Trace:	apiserver_test.go:4109
        	            				testing.go:1034
        	            				testing.go:1203
        	            				testing.go:1433
        	            				panic.go:482
        	            				testing.go:864
        	            				apiserver_test.go:4174
        	            				apiserver_test.go:3125
        	Error:      	Received unexpected error:
        	            	tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 127.0.0.1:54299->127.0.0.1:54295: write: broken pipe
        	Test:       	TestWebSessionsRenewDoesNotBreakExistingTerminalSession
FAIL
FAIL	github.com/gravitational/teleport/lib/web	30.671s
FAIL
```

**Comment by Joerger** [comment]
`TestNoReadWhenOff` was just added from the security fixes, looks like it's flaky (cc @probakowski)
```
=== RUN   TestNoReadWhenOff
    termmanager_test.go:128: 
        	Error Trace:	termmanager_test.go:128
        	Error:      	Not equal: 
        	            	expected: []byte{0x1, 0x2, 0x3}
        	            	actual  : []byte{0xda, 0x4c, 0x8b, 0xb0, 0x54, 0x43, 0x55, 0xf, 0xf0, 0x25, 0x92, 0x10, 0xf5, 0xe5, 0xb3, 0x8, 0xdf, 0x8f, 0xfc, 0xae, 0x5d, 0x63, 0xc6, 0xef, 0x93, 0xb7, 0xd4, 0x5f, 0xbb, 0x61, 0xb8, 0x38, 0x99, 0x42, 0x3c, 0x94, 0x3b, 0x7e, 0xac, 0xc4, 0x6d, 0xd0, 0x76, 0x6f, 0xa4, 0xdb, 0x37, 0xcf, 0x1e, 0xac, 0xa4, 0xfe, 0xba, 0xf8, 0xcd, 0xb7, 0x9b, 0x1c, 0xb, 0xfc, 0xdc, 0x8e, 0xcf, 0xb5, 0x79, 0xef, 0x85, 0xd4, 0xbd, 0xfd, 0x9e, 0xf0, 0xca, 0xbd, 0x14, 0xd7, 0x75, 0xa7, 0xd, 0x62, 0x37, 0x15, 0x5a, 0x99, 0x54, 0x9a, 0x26, 0x5f, 0xa, 0x82}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,8 @@
        	            	-([]uint8) (len=3) {
        	            	- 00000000  01 02 03                                          |...|
        	            	+([]uint8) (len=90) {
        	            	+ 00000000  da 4c 8b b0 54 43 55 0f  f0 25 92 10 f5 e5 b3 08  |.L..TCU..%......|
        	            	+ 00000010  df 8f fc ae 5d 63 c6 ef  93 b7 d4 5f bb 61 b8 38  |....]c....._.a.8|
        	            	+ 00000020  99 42 3c 94 3b 7e ac c4  6d d0 76 6f a4 db 37 cf  |.B<.;~..m.vo..7.|
        	            	+ 00000030  1e ac a4 fe ba f8 cd b7  9b 1c 0b fc dc 8e cf b5  |................|
        	            	+ 00000040  79 ef 85 d4 bd fd 9e f0  ca bd 14 d7 75 a7 0d 62  |y...........u..b|
        	            	+ 00000050  37 15 5a 99 54 9a 26 5f  0a 82                    |7.Z.T.&_..|
        	            	 }
        	Test:       	TestNoReadWhenOff
--- FAIL: TestNoReadWhenOff (0.00s)
```

**Comment by zmb3** [comment]
@dboslee I'm seeing a data race in `TestCAChange`

```
===================================================
OUTPUT github.com/gravitational/teleport/lib/proxy.TestCAChange
===================================================
=== RUN   TestCAChange
time="2022-06-09T15:52:02Z" level=info msg="Generating TLS certificate {0x54dc890 0xc0009f10e0 CN=test-user,O=Proxy,POSTALCODE=null,STREET= 2022-06-09 15:53:02.525403052 +0000 UTC [127.0.0.1] [] 5 []}." common_name=test-user dns_names="[127.0.0.1]" locality="[]" not_after="2022-06-09 15:53:02.525403052 +0000 UTC" org="[Proxy]" org_unit="[]" trace.component=ca
time="2022-06-09T15:52:03Z" level=info msg="Generating TLS certificate {0x54dc890 0xc000872900 CN=test-user,O=Proxy,POSTALCODE=null,STREET= 2022-06-09 15:53:03.029472405 +0000 UTC [127.0.0.1] [] 5 []}." common_name=test-user dns_names="[127.0.0.1]" locality="[]" not_after="2022-06-09 15:53:03.029472405 +0000 UTC" org="[Proxy]" org_unit="[]" trace.component=ca
time="2022-06-09T15:52:03Z" level=info msg="Generating TLS certificate {0x54dc890 0xc0001602a0 CN=test-user,O=Proxy,POSTALCODE=null,STREET= 2022-06-09 15:53:03.595219943 +0000 UTC [127.0.0.1] [] 5 []}." common_name=test-user dns_names="[127.0.0.1]" locality="[]" not_after="2022-06-09 15:53:03.595219943 +0000 UTC" org="[Proxy]" org_unit="[]" trace.component=ca
time="2022-06-09T15:52:04Z" level=info msg="Generating TLS certificate {0x54dc890 0xc0009b58c0 CN=test-user,O=Proxy,POSTALCODE=null,STREET= 2022-06-09 15:53:04.595146089 +0000 UTC [127.0.0.1] [] 5 []}." common_name=test-user dns_names="[127.0.0.1]" locality="[]" not_after="2022-06-09 15:53:04.595146089 +0000 UTC" org="[Proxy]" org_unit="[]" trace.component=ca
==================
WARNING: DATA RACE
Write at 0x00c000a18eb8 by goroutine 102:
  github.com/gravitational/teleport/lib/proxy.TestCAChange()
      /workspace/lib/proxy/client_test.go:207 +0xb0d
  testing.tRunner()
      /opt/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/go/src/testing/testing.go:1486 +0x47

Previous read at 0x00c000a18eb8 by goroutine 69:
  github.com/gravitational/teleport/lib/proxy.setupServer.func1()
      /workspace/lib/proxy/helpers_test.go:210 +0x13c
  crypto/tls.(*Conn).readClientHello()
      /opt/go/src/crypto/tls/handshake_server.go:146 +0x930
  crypto/tls.(*Conn).serverHandshake()
      /opt/go/src/crypto/tls/handshake_server.go:43 +0x6a
  crypto/tls.(*Conn).serverHandshake-fm()
      <autogenerated>:1 +0x4d
  crypto/tls.(*Conn).handshakeContext()
      /opt/go/src/crypto/tls/conn.go:1460 +0x448
  crypto/tls.(*Conn).HandshakeContext()
      /opt/go/src/crypto/tls/conn.go:1403 +0x24d
  crypto/tls.(*Conn).Handshake()
      /opt/go/src/crypto/tls/conn.go:1387 +0x232
  google.golang.org/grpc/credentials.(*tlsCreds).ServerHandshake()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/credentials/tls.go:115 +0x24e
  github.com/gravitational/teleport/lib/proxy.(*proxyCredentials).ServerHandshake()
      /workspace/lib/proxy/auth.go:47 +0x72
  google.golang.org/grpc/internal/transport.NewServerTransport()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/internal/transport/http2_server.go:145 +0xd4
  google.golang.org/grpc.(*Server).newHTTP2Transport()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:881 +0x2ce
  google.golang.org/grpc.(*Server).handleRawConn()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:838 +0xf6
  google.golang.org/grpc.(*Server).Serve.func3()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:822 +0x84

Goroutine 102 (running) created at:
  testing.(*T).Run()
      /opt/go/src/testing/testing.go:1486 +0x724
  testing.runTests.func1()
      /opt/go/src/testing/testing.go:1839 +0x99
  testing.tRunner()
      /opt/go/src/testing/testing.go:1439 +0x213
  testing.runTests()
      /opt/go/src/testing/testing.go:1837 +0x7e4
  testing.(*M).Run()
      /opt/go/src/testing/testing.go:1719 +0xa71
  main.main()
      _testmain.go:133 +0x3a9

Goroutine 69 (running) created at:
  google.golang.org/grpc.(*Server).Serve()
      /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:821 +0x704
  github.com/gravitational/teleport/lib/proxy.(*Server).Serve()
      /workspace/lib/proxy/server.go:151 +0x75
  github.com/gravitational/teleport/lib/proxy.setupServer.func3()
      /workspace/lib/proxy/helpers_test.go:234 +0x39
==================
    testing.go:1312: race detected during execution of test
--- FAIL: TestCAChange (2.71s)
===================================================
```

**Comment by zmb3** [comment]
`TestDatabaseRequest` and `TestDefaultTemplateRendering` failed due to race, but I don't see the typical race detector output:

```
=== CONT  TestDatabaseRequest
    testing.go:1312: race detected during execution of test
--- FAIL: TestDatabaseRequest (13.93s)
2022-06-16T19:06:41Z WARN [PROC:3]    Got error while cleaning up. error:[
ERROR REPORT:
Original Error: trace.aggregate stopping database foo
	rpc error: code = Canceled desc = context canceled
Stack Trace:
	/workspace/lib/srv/db/server.go:657 github.com/gravitational/teleport/lib/srv/db.(*Server).Close
	/workspace/lib/service/db.go:270 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initDatabaseService.func4
	/workspace/lib/service/service.go:1804 github.com/gravitational/teleport/lib/service.(*TeleportProcess).OnExit.func1
	/workspace/lib/service/supervisor.go:521 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
	/workspace/lib/service/supervisor.go:269 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
	/opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: stopping database foo
	rpc error: code = Canceled desc = context canceled] pid:57532.3 service/service.go:4315
```

and

```
=== CONT  TestDefaultTemplateRendering
    configtemplate_test.go:115: Expected file "teleport-user-ca.crt" for template "tls_cas" has length: 1294
    configtemplate_test.go:115: Expected file "teleport-host-ca.crt" for template "tls_cas" has length: 1289
    configtemplate_test.go:115: Expected file "teleport-database-ca.crt" for template "tls_cas" has length: 1289
    configtemplate_test.go:115: Expected file "known_hosts" for template "ssh_client" has length: 452
    configtemplate_test.go:115: Expected file "identity" for template "identity" has length: 6787
2022-06-16T19:06:41Z [PROC:2]    DEBU Broadcasting event. event:TeleportExit pid:57532.2 service/supervisor.go:376
2022-06-16T19:06:41Z [PROC:2]    DEBU Service is completed and removed. pid:57532.2 service:common.rotate service/supervisor.go:245
2022-06-16T19:06:41Z [PROC:2]    DEBU Service is completed and removed. pid:57532.2 service:fileuploadcompleter.service service/supervisor.go:245
2022-06-16T19:06:41Z [PROC:2]    DEBU Service is completed and removed. pid:57532.2 service:fileuploader.service service/supervisor.go:245
2022-06-16T19:06:41Z [AUTH:2]    INFO Shutting down immediately. pid:57532.2 service/service.go:1769
    testing.go:1312: race detected during execution of test
--- FAIL: TestDefaultTemplateRendering (13.98s)
==================================================
```

  https://console.cloud.google.com/cloud-build/builds/f5d388e0-f46e-4c7c-b6fd-5035581f4714;step=0?project=ci-account

**Comment by ibeckermayer** [comment]
`TestAgentForwardPermission`: https://github.com/gravitational/teleport/issues/13565

**Comment by ibeckermayer** [comment]
`TestSSHAccessRequest`: https://github.com/gravitational/teleport/issues/13566

**Comment by ibeckermayer** [comment]
`TestIntegrations/RotateTrustedClusters`: https://github.com/gravitational/teleport/issues/13569

**Comment by ibeckermayer** [comment]
`TestHSMDualAuthRotation`: https://github.com/gravitational/teleport/issues/13635

**Comment by ibeckermayer** [comment]
`TestIntegrations/Disconnection`: https://github.com/gravitational/teleport/issues/13699

**Comment by ibeckermayer** [comment]
`TestLockInForce`: https://github.com/gravitational/teleport/issues/13770

**Comment by ibeckermayer** [comment]
`TestAgentForward`: https://github.com/gravitational/teleport/issues/13796

**Comment by ibeckermayer** [comment]
`TestHSMRotation`: https://github.com/gravitational/teleport/issues/13845

**Comment by ibeckermayer** [comment]
`TestProxyClientDisconnectDueToIdleConnection`: https://github.com/gravitational/teleport/issues/13881

**Comment by ibeckermayer** [comment]
`TestOpenExecSessionSetsSession`: https://github.com/gravitational/teleport/issues/13882

**Comment by ibeckermayer** [comment]
`TestProxyTunnelStrategyAgentMesh`: https://github.com/gravitational/teleport/issues/13631

**Comment by ibeckermayer** [comment]
`TestHSMRotation`: https://github.com/gravitational/teleport/issues/13599

**Comment by ibeckermayer** [comment]
`TestSiteNodeConnectInvalidSessionID`: https://github.com/gravitational/teleport/issues/13889

**Comment by ibeckermayer** [comment]
`TestProxyRoundRobin`: https://github.com/gravitational/teleport/issues/13938

**Comment by ibeckermayer** [comment]
`TestTerminalRequireSessionMfa`: https://github.com/gravitational/teleport/issues/13944

**Comment by ibeckermayer** [comment]
`TestDatabaseRequest`: https://github.com/gravitational/teleport/issues/13950 (original comment [here](https://github.com/gravitational/teleport/issues/9492#issuecomment-1158058671))

**Comment by ibeckermayer** [comment]
`TestInteractiveSession/Stop`: https://github.com/gravitational/teleport/issues/13952

**Comment by ibeckermayer** [comment]
`TestTLSConfiguration/insecure_ignores_incorrect_CN/mongodb`: https://github.com/gravitational/teleport/issues/13953

**Comment by ibeckermayer** [comment]
@espadolini noticed something that's potentially generally relevant in the output of the logs I got from running the `RotateTrustedClusters` test using `count=10` locally like

```
go test -count=10 -run TestIntegrations/RotateTrustedClusters -v github.com/gravitational/teleport/integration
```

It's not uncommon to see processes from previous tests hanging around in the logs, e.g.
```
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:12","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.12","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:17","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.17","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:2","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.2","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:22","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.22","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-main","component":"proc:34","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.34","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:32","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.32","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:7","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.7","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:27","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.27","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:22","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.22","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-main","component":"proc:34","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.34","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:17","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.17","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:12","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.12","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:2","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.2","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:32","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.32","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:7","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.7","timestamp":"2022-06-29T11:20:01-04:00"}
{"caller":"service/supervisor.go:376","cluster":"rotate-aux","component":"proc:27","event":"TeleportDegraded","level":"debug","message":"Broadcasting event.","pid":"58352.27","timestamp":"2022-06-29T11:20:01-04:00"}
```

This section of the logs has `proc`s 12, 17, 2, 22, 34, etc all talking at once. We expect 5 procs per test run (e.g. 50 total for `count=10`), but because those runs are serialized (afaik), those process numbers suggest that the old `TeleportProcess`es are all hanging around. It's unclear how much this might contribute to general flakiness and/or our inability to `t.Parallel()`ize certain tests, but it seems like it could be relevant.

**Comment by zmb3** [comment]
`TestALPNProxyHTTPProxyBasicAuthDial`: #14143

**Comment by ibeckermayer** [comment]
`TestHSMMigrate`: https://github.com/gravitational/teleport/issues/14172

**Comment by ibeckermayer** [comment]
`TestRedisTransaction`: https://github.com/gravitational/teleport/issues/14297

**Comment by ravicious** [comment]
`TestIgnorePuTTYSimpleChannel`: https://github.com/gravitational/teleport/issues/14368

**Comment by nklaassen** [comment]
`TestBot_Run_CARotation`: https://github.com/gravitational/teleport/issues/14471

**Comment by nklaassen** [comment]
`TestFnCacheSanity`: https://github.com/gravitational/teleport/issues/14534

**Comment by nklaassen** [comment]
`TestAgentStart`: https://github.com/gravitational/teleport/issues/14553

**Comment by nklaassen** [comment]
`TestNormalOperation`: https://github.com/gravitational/teleport/issues/14554

**Comment by nklaassen** [comment]
`TestFnCacheCancellation`: https://github.com/gravitational/teleport/issues/14556

**Comment by ravicious** [comment]
`TestTokens`: #14737 (`tool/tctl/common/token_command_test.go`)

**Comment by zmb3** [comment]
`TestSemaphoreLock`: https://github.com/gravitational/teleport/issues/14842

**Comment by zmb3** [comment]
`TestIntegrations/ListResourcesAcrossClusters`: #15051

**Comment by zmb3** [comment]
`TestIntegrations/SessionRecordingModes/StrictMode`: #15097

**Comment by nklaassen** [comment]
`failed to parse existing kubeconfig` tsh test flakiness #15490

**Comment by ibeckermayer** [comment]
`TestMain` (sshserver_test.go): https://github.com/gravitational/teleport/issues/15520

**Comment by GavinFrazar** [comment]
`TestAllowedUsers`: https://github.com/gravitational/teleport/issues/15656

**Comment by ibeckermayer** [comment]
`TestRoleDeletionDrift`: https://github.com/gravitational/teleport/issues/15815

**Comment by Joerger** [comment]
`TestWebSessionsRenewDoesNotBreakExistingTerminalSession`: https://github.com/gravitational/teleport/issues/15816

**Comment by ibeckermayer** [comment]
`tbot` race condition: https://github.com/gravitational/teleport/issues/15843

**Comment by espadolini** [comment]
`TestPingConnection`: https://github.com/gravitational/teleport/issues/15896

**Comment by strideynet** [comment]
`TestTeleportClient_Login_local/OTP_device_login_with_hijack`: https://github.com/gravitational/teleport/issues/16037

**Comment by GavinFrazar** [comment]
`TestHandlerConnectionUpgrade` data race: #16128

**Comment by ibeckermayer** [comment]
`TestGlobalRequestRecordingProxy`: https://github.com/gravitational/teleport/issues/16138

**Comment by ibeckermayer** [comment]
`TestAppAccess/TestAppInvalidateAppSessionsOnLogout`: https://github.com/gravitational/teleport/issues/16200

**Comment by nklaassen** [comment]
`TestRootScript`: https://github.com/gravitational/teleport/issues/16908

**Comment by tigrato** [comment]
`TestIntegrations/AuditOn`: #17224

**Comment by nklaassen** [comment]
`TestWebAgentForward`: #17918

**Comment by nklaassen** [comment]
`TestIntegrations/TrustedTunnelNode`: #17824

**Comment by ibeckermayer** [comment]
FTR its no longer expected that this issue be updated with each new flaky test. To easily see/search all the flaky tests, bookmark [this url](https://twitter.com/rdnxyz/status/1586056398609096706).
